deepchecks/vision/dataset.py

```diff
@@ -0,0 +1,140 @@
+from copy import copy
+from enum import Enum
+from collections import Counter
+from typing import Callable
+
+from torch.utils.data import DataLoader
+from torch import cat
+import logging
+
+from deepchecks.core.errors import DeepchecksValueError
+
+logger = logging.getLogger('deepchecks')
+
+__all__ = ['TaskType', 'VisionDataset']
+
+
+class TaskType(Enum):
+    """Enum containing supported task types."""
+
+    CLASSIFICATION = 'classification'
+    OBJECT_DETECTION = 'object_detection'
+    SEMANTIC_SEGMENTATION = 'semantic_segmentation'
+
+
+class VisionDataset:
```

Comment:
TODO: add docstring. Should elaborate also on required label format after label_transformer